src/promptflow-evals/promptflow/evals/synthetic/simulator/_rai_rest_client/rai_client.py

```diff
@@ -0,0 +1,70 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+import logging
+import os
+from typing import Any
+
+from promptflow.evals.synthetic.simulator._model_tools.models import AsyncHTTPClientWithRetry
+
+api_url = None
+if "rai_svc_url" in os.environ:
+    api_url = os.environ["rai_svc_url"]
+    api_url = api_url.rstrip("/")
+    print(f"Found rai_svc_url in environment variable, using {api_url} for rai service endpoint.")
+
+
+class RAIClient:  # pylint: disable=client-accepts-api-version-keyword
+    # pylint: disable=missing-client-constructor-parameter-credential, missing-client-constructor-parameter-kwargs
+    def __init__(self, ml_client: Any, token_manager: Any) -> None:
+        self.ml_client = ml_client
+        self.token_manager = token_manager
+
+        self.contentharm_parameters = None
+        self.jailbreaks_dataset = None
+
+        if api_url is not None:
+            host = api_url
+        else:
+            host = self.ml_client.jobs._api_url
+
+        self.api_url = (
+            f"{host}/"
+            + f"raisvc/v1.0/subscriptions/{self.ml_client.subscription_id}/"
+            + f"resourceGroups/{self.ml_client.resource_group_name}/"
+            + f"providers/Microsoft.MachineLearningServices/workspaces/{self.ml_client.workspace_name}/"
+        )
+
+        self.parameter_json_endpoint = self.api_url + "simulation/template/parameters"

```

Comment:
Please use urllib to avoid any trailing / getting missing accidentally

Reply:
We'll fix this in the next iteration